submissions/chernetskyi8704/Frogger/app.js

```diff
@@ -0,0 +1,169 @@
+const TILE_SIZE = {
+  height: 83,
+  width: 101,
+};
+
+const FILED_SIZE = {
+  height: TILE_SIZE.height * 6,
+  width: TILE_SIZE.width * 5,
+};
+
+const FIELD_SIZE_LIMIT = {
+  height: FILED_SIZE.height - TILE_SIZE.height,
+  width: FILED_SIZE.width - TILE_SIZE.width,
+};
+
+const PLAYER_CONF = {
+  initialPosition: {
+    x: 202,
+    y: 401,
+  },
+  speed: {
+    min: 72,
+    max: 150,
+  },
+  step: 15,
+  minDistanceToCollision: {
+    height: 75,
+    width: 80,
+  },
+  sprite: "images/char-boy.png",
+};
+
+const ENEMY_CONF = {
+  initialPosition: {
+    x: -TILE_SIZE.width,
+    y: {
+      enemy1: 63,
+      enemy2: 147,
+      enemy3: 230,
+    },
+  },
+  sprite: "images/enemy-bug.png",
+};
+
+let score = 0;
+// Enemies our player must avoid
+const Enemy = function (x, y) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+  this.x = x;
+  this.y = y;
+  this.speed = this.randomSpeed();
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+  this.sprite = ENEMY_CONF.sprite;
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.randomSpeed = function () {
+  return (
+    Math.floor(Math.random() * PLAYER_CONF.speed.max) + PLAYER_CONF.speed.min
+  );
+};
+
+Enemy.prototype.update = function (dt) {
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+  this.x += this.speed * dt;
+
+  if (this.x > FILED_SIZE.width) {
+    this.x = ENEMY_CONF.initialPosition.x;
+    this.speed = this.randomSpeed();
+  }
+
+  this.collision();
+};
+
+Enemy.prototype.collision = function () {
+  if (
+    this.x + PLAYER_CONF.minDistanceToCollision.width > player.x &&
+    this.x < player.x + PLAYER_CONF.minDistanceToCollision.width &&
+    this.y + PLAYER_CONF.minDistanceToCollision.height > player.y &&
+    this.y < player.y + PLAYER_CONF.minDistanceToCollision.height
+  ) {
+    score = 0;
+    player.resetInitialPosition();
+  }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+const Player = function (x, y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = PLAYER_CONF.sprite;
+};
+
+Player.prototype.update = function () {};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.resetInitialPosition = function () {
+  player.x = PLAYER_CONF.initialPosition.x;
+  player.y = PLAYER_CONF.initialPosition.y;
```

Comment:
My mistake is that I refer to a specific class instance in the method, it would be more correct to write this.x\this.y, because, for example, if we create player2 using the Player constructor function with different initial coordinates and call the resetInitialPosition() method for this player2, the method will not work.

If the method would write x = PLAYER_CONF.initialPosition.x; it would not make sense, it would just mean that this method writes a value to the property, it would be like let x = PLAYER_CONF.initialPosition.x;